backend/src/v5/processors/teamspaces/projects/models/commons/tickets.groups.js

```diff
@@ -40,14 +57,23 @@ const getObjectArrayFromRules = async (teamspace, project, model, revId, rules)
 		}
 	}
 
-	const [
-		{ matched, unwanted },
-		idToMeshes,
-	] = await Promise.all([
-		getMetadataByRules(teamspace, project, model, revision, rules, { parents: 1 }),
-		getIdToMeshesMapping(teamspace, model, revision),
-	]);
+	const projection = { parents: 1,
+		...(convertTo3dRepoIds ? {} : { metadata: {
+			$elemMatch: { $or: Object.values(externalIdNamesToKeys).flat().map((n) => ({ key: n })) } } }) };
 
+	const { matched, unwanted } = await getMetadataByRules(teamspace, project, model, revision, rules, projection);
+
+	if (!convertTo3dRepoIds && matched.some((m) => m.metadata)) {
+		const wantedIds = [...new Set(matched.map(({ metadata }) => metadata[0].value))];
+		const unwantedIds = [...new Set(unwanted.map(({ metadata }) => metadata[0].value))];
+
+		unwantedIds.filter((id) => (wantedIds.includes(id))).forEach((id) => delete wantedIds[id]);
+
+		const externalIdName = getExteralidNameFromMetadata(matched);
+		return { container: model, [externalIdName]: wantedIds };
+	}
+
+	const idToMeshes = await getIdToMeshesMapping(teamspace, model, revision);
```

Comment:
Use helper function that I've suggested previously [get the difference](https://github.com/3drepo/3drepo.io/blob/master/backend/src/v5/utils/helper/arrays.js#L23)

Reply:
something like this or not at all ? 🤔 

![image](https://github.com/3drepo/3drepo.io/assets/71635617/236d92d3-f1bc-413c-a9a4-a7d84f87e494)

Reply:
errr, just leave it, i'll have another look on #4641